### PR TITLE
[Feature] Add multiple price sources to collateral oracle

### PR DESCRIPTION
**Pull request by csanti**
### Overview

We want to add to the collateral oracle 2 additional price sources:
- Fixed price: When registering the collateral the owner specifies a fixed rate.
- Band oracle: Query the price directly from band protocol's oracle.

Because now we have collaterals that have completely different logic when obtaining the price, we need to substitute the current generic `query_request` for a `source_type`:

```
pub enum SourceType {
    TerraOracle {
        terra_oracle_query: Binary,
    },
    BandOracle {
        band_oracle_query: Binary,
    },
    FixedPrice {
        price: Decimal,
    },
    Terraswap {
        terraswap_query: Binary,
    }
} 
```

### Querying price from Band Oracle

Band oracle has the peculiarity of returning the price as a `Uint128` instead of a `Decimal` [(Band oracle doc)](https://github.com/bandprotocol/cosmwasm-std-reference-doc#struct-types) The `Uint128` contains the price multiplied by `1e18`.

Parsing the `Uint128` can be tricky because the conversion operation, would result in a `uint128` overflow. If instead we use 256 variant, it still could have the possibility of overflowing.

Therefore, in this implementation I opted to parse it as a string to safely move the decimal point.

### CODE REVIEW

File: contracts/mirror_collateral_oracle/src/querier.rs [hunk 1]
```diff
@@ -1,34 +1,53 @@
...
+    pub rate: Decimal,
+}
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct TerraswapResponse {
     // terraswap queries return pool assets
-    pub assets: Option<[Asset; 2]>,
+    pub assets: [Asset; 2],
+}
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct BandOracleResponse {
```
Comment (YunSuk-Yeo) [inline contracts/mirror_collateral_oracle/src/querier.rs:22]: I have example query here https://lcd.terra.dev/wasm/contracts/terra1mwshjfdth2xtpx7gvnwpzhks2clr9z7h9m0e0f/store?query_msg=%7B%22get_reference_data%22:%7B%22base_symbol%22:%22AAPL%22,%22quote_symbol%22:%22USD%22%7D%7D

**Review by YunSuk-Yeo — CHANGES_REQUESTED** [review]
LGTM, changing the Binary to SourceType looks very good.

band & terra oracle query has updated time data, so we can use that data to give return error response when it has not been updated for too long time.

File: contracts/mirror_collateral_oracle/src/querier.rs (hunk 1 above)
Comment (csanti) [reply contracts/mirror_collateral_oracle/src/querier.rs:23]: Added requested changes

**Comment by csanti** [comment]
Added the requested changes. Now collateral oracle returns last_updated parameter. The price age is checked by mint contract.

**Review by YunSuk-Yeo — APPROVED** [review]

**Pull request merged**